CourseSketchProjects/IdentityServer/src/test/java/coursesketch/database/identity/IdentityManagerTest.java

```diff
@@ -0,0 +1,922 @@
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by dtracers on 10/8/2015.
+ */
+@RunWith(PowerMockRunner.class)
+public class IdentityManagerTest {
+
+    @Rule
+    public FongoRule fongo = new FongoRule();
+
+    private static final String VALID_USERNAME = "Valid username";
+    private static final String VALID_REGISTRATION_KEY = "VALID KEY YO";
+    private static final String INVALID_REGISTRATION_KEY = "NOT VALID KEY YO";
+    private static final String INVALID_USERNAME = "NOT VALID USERNAME";
+
+    public static final School.ItemType INVALID_ITEM_TYPE = School.ItemType.LECTURE;
+    public static final School.ItemType VALID_ITEM_TYPE = School.ItemType.COURSE;
+    public static final School.ItemType VALID_ITEM_CHILD_TYPE = School.ItemType.ASSIGNMENT;
+
+    public static final String INVALID_ITEM_ID = new ObjectId().toHexString();
+    public static final String VALID_ITEM_CHILD_ID = new ObjectId().toHexString();
+    public static final String VALID_ITEM_ID = new ObjectId().toHexString();
+
+    public static final String VALID_GROUP_ID = new ObjectId().toHexString();
+    public static final String INVALID_GROUP_ID = new ObjectId().toHexString();
+
+    public static final String TEACHER_AUTH_ID = new ObjectId().toHexString();
+    public static final String STUDENT_AUTH_ID = new ObjectId().toHexString();
+    public static final String MOD_AUTH_ID = new ObjectId().toHexString();
+
+    public static final String TEACHER_USER_ID = new ObjectId().toHexString();
+    public static final String STUDENT_USER_ID = new ObjectId().toHexString();
+    public static final String MOD_USER_ID = new ObjectId().toHexString();
+
+    // this user id is not in the db
+    public static final String NO_ACCESS_ID = new ObjectId().toHexString();
+
+    public DB db;
+
+    public IdentityManager identityManager;
+    Authenticator dbAuthChecker;
+    @Mock
+    private AuthenticationChecker authChecker;
+    @Mock
+    private AuthenticationOptionChecker optionChecker;
+
+    @Before
+    public void before() throws UnknownHostException {
+
+        db = fongo.getDB(); // new MongoClient("localhost").getDB("test");
+        identityManager = new IdentityManager(db);
+
+        try {
+            // general rules
+            AuthenticationHelper.setMockPermissions(authChecker, null, null, null, null, Authentication.AuthResponse.PermissionLevel.NO_PERMISSION);
+
+            when(optionChecker.authenticateDate(any(AuthenticationDataCreator.class), anyLong()))
+                    .thenReturn(false);
+
+            when(optionChecker.isItemPublished(any(AuthenticationDataCreator.class)))
+                    .thenReturn(false);
+
+            when(optionChecker.isItemRegistrationRequired(any(AuthenticationDataCreator.class)))
+                    .thenReturn(true);
+        } catch (DatabaseAccessException e) {
```

Comment:
chance to:
`} catch (DatabaseAccessException | AuthenticationException e) {..}`
since you do the exact same thing in both